src/Job/DeveloperUpdate.php

```diff
@@ -19,41 +19,103 @@
 
 namespace Drupal\apigee_edge\Job;
 
-use Drupal\apigee_edge\Entity\DeveloperInterface;
+use Drupal\apigee_edge\Entity\Developer;
+use Drupal\apigee_edge\Entity\FieldableEdgeEntityUtilityTrait;
 
 /**
- * A job to update a developer in Edge.
+ * A job to update an Apigee Edge developer based on a Drupal user.
  */
 class DeveloperUpdate extends EdgeJob {
 
+  use FieldableEdgeEntityUtilityTrait;
+
   /**
-   * A developer who is being updated.
+   * The email of the developer/user.
    *
-   * @var \Drupal\apigee_edge\Entity\DeveloperInterface
+   * @var string
    */
-  protected $developer;
+  protected $mail;
 
   /**
-   * {@inheritdoc}
+   * Whether the Apigee Edge developer should be updated.
+   *
+   * @var bool
    */
-  public function __construct(DeveloperInterface $developer) {
+  protected $executeUpdate = FALSE;
+
+  /**
+   * DeveloperUpdate constructor.
+   *
+   * @param string $mail
+   *   The email of the developer/user.
+   */
+  public function __construct(string $mail) {
     parent::__construct();
-    $this->developer = $developer;
+    $this->mail = $mail;
   }
 
   /**
    * {@inheritdoc}
    */
   protected function executeRequest() {
-    $this->developer->save();
+    /** @var \Drupal\apigee_edge\Entity\DeveloperInterface $developer */
+    $developer = Developer::load($this->mail);
+    /** @var \Drupal\user\UserInterface $account */
+    $account = user_load_by_mail($this->mail);
+
+    if ($developer->getFirstName() !== $account->get('first_name')->value) {
+      $developer->setFirstName($account->get('first_name')->value);
+      $this->executeUpdate = TRUE;
+    }
+
+    if ($developer->getLastName() !== $account->get('last_name')->value) {
+      $developer->setLastName($account->get('last_name')->value);
+      $this->executeUpdate = TRUE;
+    }
+
+    $user_fields_to_sync = \Drupal::config('apigee_edge.sync')->get('user_fields_to_sync');
+    if (!empty($user_fields_to_sync)) {
+      /** @var \Drupal\apigee_edge\FieldStorageFormatManager $format_manager */
+      $format_manager = \Drupal::service('plugin.manager.apigee_field_storage_format');
+      foreach ($user_fields_to_sync as $field) {
+        $field_definition = $account->getFieldDefinition($field);
+        if (!isset($field_definition)) {
+          $this->recordMessage(t('Skipping @mail developer update, because the field @field does not exist.', [
```

Comment:
s/because the field @field does/because %field field does/g

Reply:
Are you skipping a developer entity _completely_ just because one field does not exist? I think you are not. Just the displayed message is incorrect. You are skipping synching a field value of an entity because field does not exist.